contracts/multisigs/GnosisSafeMultisig.sol

```diff
@@ -33,15 +40,22 @@ contract GnosisSafeMultisig {
     }
 
     /// @dev Parses (unpacks) the data to gnosis safe specific parameters.
+    /// @notice If the provided data is not empty, its length must be at least 144 bytes to be parsed correctly.
     /// @param data Packed data related to the creation of a gnosis safe multisig.
     function _parseData(bytes memory data) internal pure
         returns (address to, address fallbackHandler, address paymentToken, address payable paymentReceiver,
             uint256 payment, uint256 nonce, bytes memory payload)
     {
-        if (data.length > 0) {
-            uint256 dataSize = data.length;
+        uint256 dataLength = data.length;
+        if (dataLength > 0) {
+            // Check for the correct data length
+            if (dataLength < DEFAULT_DATA_LENGTH) {
+                revert IncorrectDataLength(DEFAULT_DATA_LENGTH, dataLength);
+            }
```

Comment:
If the data is not zero, we need to check for it to be at least 144 bytes length, otherwise we are guaranteed to not parse it correctly